File: Scripts/server/concurrent-computing/c-sharp/concurrent-computing.cs
using System;
using System.Threading;

class Concurrent {

private static volatile int r = 1;

public static int executeTask(int i) {

	Thread t = new Thread(new ParameterizedThreadStart(WriteText));
	t.Start("Enjoy");
 
	t = new Thread(new ParameterizedThreadStart(WriteText));
	t.Start("Rosetta");
 
	t = new Thread(new ParameterizedThreadStart(WriteText));
	t.Start("Code");	
	return i + 1;
}

static void Main(string[] args)
{
	for ( int i = 0; i < 1000000; ++i ) {
		r = executeTask(i + r);
	}
}
 
private static void WriteText(object p)
{
	Thread.Sleep(1);
	Console.WriteLine(p);
}
}

File: scripts/volatile_and_force_compiler_task/concurrent-computing/visual-basic-.net/concurrent-computing.vb
Imports System.Threading

Module Module1
   Public rnd As New Random

   Sub Main()
       

    For value As Integer = 0 To 1000000

       Dim t1 As New Thread(AddressOf Foo)
       Dim t2 As New Thread(AddressOf Foo)
       Dim t3 As New Thread(AddressOf Foo)

       t1.Start(value.ToString() + "Enjoy")
       t2.Start(value.ToString() + "Rosetta")
       t3.Start(value.ToString() + "Code")

       t1.Join()
       t2.Join()
       t3.Join()
    Next

   End Sub

   Sub Foo(ByVal state As Object)
       Thread.Sleep(rnd.Next(1))
       Console.WriteLine(state)
   End Sub

End Module
